proposals/trestle-sdk-cli-decoupling.md

```diff
@@ -0,0 +1,160 @@
+
+Initial discussion occurred in the OSCAL Compass [community meeting](https://www.youtube.com/watch?v=uUGv3HXlTrI&t=19s  ) on June 14, 2024 pertaining to the reusability of `Rule` logic to support this [issue](https://github.com/oscal-compass/compliance-trestle/issues/1475). An [issue](https://github.com/oscal-compass/community/issues/28) in the community repo was subsequently submitted.
+
+## Goals
+
+- `compliance-trestle` continues to contain the `trestle` CLI logic
+- `compliance-trestle-python` is created as the Python SDK repository
+- `compliance-trestle` and other OSCAL Compass projects migration to the new SDK for applicable logic
+
+## Non-Goals
+
+- Fundamentally changing the architecture or design of the SDK or CLI is out of scope.
+- Expanding the SDK to other programming languages is out of scope (future consideration).
+
+## Proposal
+
+The solution proposed in this document is to move certain pieces of the codebase located in the [compliance-trestle](https://github.com/oscal-compass/compliance-trestle) repository to a new repository containing a Python-based SDK for interaction with OSCAL documents.
+
+Below is high-level description of what would be moved:
+
+**OSCAL Classes**: The classes that represet the OSCAL data model.  
+**Core Functionality**: Contains the fundamental APIs for interacting with OSCAL objects.  
+**Abstractions**: Abstractions for common operations like resolution or core OSCAL validation.  
+  
+</details>
+
+## Design Details
+
+Below is diagram with existing `trestle` logic that is proposed to be part of the SDK. Links to where this type of logic exists currently are included:
+
+```mermaid
+classDiagram
+    OSCALBaseModel <|-- Catalog
+    OSCALBaseModel <|-- Profile
+    OSCALBaseModel <|-- ComponentDefinitions
+    OSCALBaseModel <|-- SSP
+    OSCALBaseModel <|-- AssessmentPlan
+    OSCALBaseModel <|-- AssessmentResult
+    OSCALBaseModel <|-- POAM
+    class OSCALBaseModel{
+      +Config
+      +write()
+      +read()
+    }
+    class Catalog{
+      +CatalogAPI
+    }
+    class Validator{
+      +isValid(OSCALBaseModel)
+    }
+    class Resolver{
+      +Fetcher
+      +import()
+      +merge()
+      +modify()
+      +prune()
+    }
+```
+
+### Relevant Links
+
+ - OSCAL Classes: https://github.com/oscal-compass/compliance-trestle/tree/develop/trestle/oscal
+  - Core Logic: https://github.com/oscal-compass/compliance-trestle/blob/develop/trestle/core/models/
+  - Abstractions: 
+    - https://github.com/oscal-compass/compliance-trestle/tree/develop/trestle/core/resolver
+    - https://github.com/oscal-compass/compliance-trestle/blob/dfe892936e5960ad64f6f387dbe5918314049e89/trestle/core/validator.py
+
+## Impacts / Key Questions
+
+Some of my main questions around this proposal are the following:
+
+- What will the contribution process look like for logic in `trestle`?
+- Will the SDK be maintained by the `compliance-trestle` maintainer or by a new set of maintainers?
+
```

Comment:
`Can you please share what you are looking for in terms of community support? This was shared at the last community meeting and what I perceived is that with some updates to the implementation and more understanding of the level of effort, this was supported by the group.`

If there is community support for this proposal please move forward. I've been known to be wrong. I will keep quiet now.